Automattic/wp-calypso
# 2FA: update reauth component to properly render children when reauth successfully

#### Proposed Changes
In this [PR](https://github.com/Automattic/wp-calypso/pull/70598) we introduce a new state variable `isTwoStepInitialized` to determine if two step is initialised before trying to render it's children. However we needed to update `isTwoStepInitialized` again when a new change is emitted otherwise `isTwoStepInitialized` can be false if the render happens before initialisation is completed from the backend.

* This PR removes the derived state that's causing the issue and access updated values directly from props. See [comment](https://github.com/Automattic/wp-calypso/pull/71017#issuecomment-1350428871) 

To **reproduce** this issue we need to wrap a component that depends on 2FA as child of `<ReauthRequired/>`
apply diff

```diff
diff --git a/client/me/connected-applications/index.jsx b/client/me/connected-applications/index.jsx
index 028a423571..7ae9cd6b58 100644
--- a/client/me/connected-applications/index.jsx
+++ b/client/me/connected-applications/index.jsx
@@ -95,8 +95,9 @@ class ConnectedApplications extends PureComponent {
 						{ translate( 'Connected Applications' ) }
 					</HeaderCake>
 				) }
-
-				{ this.renderConnectedApps() }
+				<ReauthRequired twoStepAuthorization={ twoStepAuthorization }>
+					{ () => this.renderConnectedApps() }
+				</ReauthRequired>
 			</Fragment>
 		);
 	}
@@ -112,7 +113,6 @@ class ConnectedApplications extends PureComponent {
 					path="/me/security/connected-applications"
 					title="Me > Connected Applications"
 				/>
-				<ReauthRequired twoStepAuthorization={ twoStepAuthorization } />
 
 				<FormattedHeader brandFont headerText={ translate( 'Security' ) } align="left" />
```

Go to `me/security/connected-applications` and refresh the page. Notice the list of applications are not shown
![image](https://user-images.githubusercontent.com/47489215/207473211-8fe9c2c1-4af8-49b2-979d-a0c04b708478.png)
this is because `isTwoStepInitialized` was set to false but wasn't updated when `twoStepAuthorization` finish initialising and emit the change.


#### Testing Instructions
1. Checkout this branch.
2. Apply the diff from above
3. Go to `me/security/connected-applications` and ensure list of connected apps are now shown. 
4. In chrome delete `two_step` cookie and ensure the `RequiredAuth` dialog is shown


*

#### Pre-merge Checklist





Related to #
